### PR TITLE
HTML template support in file upload

### PR DESCRIPTION
**Pull request by cazitouni**
Hello, 

This PR add the suport to upload html templates into the info-templates folder in the already existing file uploader from the theme plugin. 

Do not hesitate to inform me if you have any remarks. 

Kind regards, Clément.

### CODE REVIEW

File: plugins/themes/controllers/files_controller.py [hunk 1]
```diff
@@ -43,25 +43,39 @@ def __init__(self, app, handler):
...
+        )
 
         self.app = app
         self.handler = handler
         self.template_dir = "plugins/themes/templates"
 
         config_handler = handler()
         self.resources_path = config_handler.config().get("qgs_resources_path")
-
+        self.info_templates_path = config_handler.config().get("info_templates_path")
```
Comment (manisandro) [inline plugins/themes/controllers/files_controller.py:64]: Please document the setting in the schema definition [1].

[1] https://github.com/qwc-services/qwc-admin-gui/blob/master/schemas/qwc-admin-gui.json#L107

Comment (manisandro) [reply plugins/themes/controllers/files_controller.py:64]: And please also add the setting in the check here [1]

[1] https://github.com/qwc-services/qwc-admin-gui/blob/master/plugins/themes/__init__.py#L17

Comment (cazitouni) [reply plugins/themes/controllers/files_controller.py:64]: Should be good now

**Comment by manisandro** [comment]
Thanks - so as I see this is merely the file upload part, no configuration is performed in the tenantConfig.json right?

**Comment by cazitouni** [comment]
Hello @manisandro, 

Yes I've made this part first and I was thinking about submitting the new controller etc.. for html templates in a separate PR, but I need a few more days to find the time to re-read my code. 

Would you prefer I commit everything in this one ? 

Thanks, regards, Clément.

**Comment by manisandro** [comment]
It might be nice to also have the plugin write the `resources` config section of the `featureinfo` service configuration in `tenantConfig.json` according to the selected template. But I can also start merging this part if you prefer.

**Comment by cazitouni** [comment]
Hello, 

Yes, I've already made all of that. 
I just need to find a couple hours in my work schedule to test it better and proofread it.

I will commit everything in this one then it'll be easier. I'll try to make it by tommorow,  I'll let you know.

Thanks for your answers, Regards, Clément.

**Comment by cazitouni** [comment]
Hello @manisandro, 

New commit should include everything.
I've tested on my config and everything should works, but I may have made some mistakes sot let me know if it's ok for you.

In adition if it's ok we should also need to modify default docker's compose config to include the info-templates volumes and the admin gui documentation to include the new variables(qgis-server-url and info_templates path).

Thanks in advance, regards, Clément.

**Comment by cazitouni** [comment]
> Looks good overall. As a general remark, I'd use `info_template` rather than just `template` as the base wording throughout, as `template` is too generic.

Ok i'll look into it

**Comment by cazitouni** [comment]
It should be good vor the variables names now.
Also, I was thinking saving the tenant config in a separate .bak file each time a template is associated isn't a bit too much ? 
It's safer but it'll make a lot of .bak files in the long run.

**Comment by cazitouni** [comment]
Hello @manisandro, 

I allow myself to solicit you again about this PR.

Thanks you in advance, regards, Clément.

**Comment by manisandro** [comment]
Sorry, fell off my radar. Thanks for your work!

**Pull request merged**